crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -135,12 +137,122 @@ impl Persistence {
             .await
             .map_err(Error::DbError)
     }
+
+    /// Get auction data.
+    pub async fn get_auction(
+        &self,
+        auction_id: domain::auction::Id,
+    ) -> Result<domain::settlement::Auction, error::Auction> {
+        let mut ex = self.postgres.pool.begin().await.context("begin")?;
+
+        let deadline = {
+            let scores = database::settlement_scores::fetch(&mut ex, auction_id)
+            .await
+            .context("fetch scores")?
+            // if score is missing, no competition / auction exist for this auction_id
+            .ok_or(error::Auction::Missing)?;
+
+            (scores.block_deadline as u64).into()
+        };
+
+        let prices = {
+            let db_prices = database::auction_prices::fetch(&mut ex, auction_id)
+                .await
+                .context("fetch auction prices")?;
+
+            let mut prices = HashMap::new();
+            for price in db_prices {
+                let token = eth::H160(price.token.0).into();
+                let price = big_decimal_to_u256(&price.price)
+                    .ok_or(domain::auction::InvalidPrice)
+                    .and_then(|p| domain::auction::Price::new(p.into()))
+                    .map_err(error::Auction::Price)?;
+                prices.insert(token, price);
+            }
+            prices
+        };
+
+        let fee_policies = {
+            let orders = database::auction_orders::fetch(&mut ex, auction_id)
+                .await
+                .context("fetch auction orders")?
+                .ok_or(error::Auction::Missing)?
+                .into_iter()
+                .map(|order| domain::OrderUid(order.0));
+
+            let mut fee_policies = HashMap::new();
+            for order in orders {
+                fee_policies.insert(order, vec![]);
+                if database::orders::read_order(&mut ex, &ByteArray(order.0))
+                    .await
+                    .context("fetch order")?
+                    .is_some()
+                {
+                    let quote = database::orders::read_quote(&mut ex, &ByteArray(order.0))
+                        .await
+                        .context("fetch quote")?
+                        .map(dto::quote::into_domain)
+                        .transpose()
+                        .map_err(|_| error::Auction::DbConversion("quote overflow"))?
+                        .ok_or(error::Auction::MissingQuote)?;
+
+                    let policies =
+                        database::fee_policies::fetch(&mut ex, auction_id, ByteArray(order.0))
+                            .await
+                            .context("fetch fee policies")?;
```

Comment:
3*N queries are made here, where N is the number of orders. How about making just 3 queries in total by reading all the orders at once, then all the quotes, and, in the end, all the policies? 

nit: Orders and quotes could be joined in the SQL, which remains to be a simple query.

Reply:
Just realized that due to `sqlx` limitations, fetching all the fee policies at once won't be possible without postgres-specific queries: https://github.com/cowprotocol/services/pull/2809#discussion_r1675527726
But if we might have ~2k orders, IMO, it should still be considered to make 3 queries instead of 2k.

Reply:
Transformed to 3 queries 👍

Reply:
Joining orders and quotes is a more involving change outside of scope of PR, should be tackled when we refactor solvable orders cache.